src/assets/scripts/client/aircraft/FlightManagementSystem/RouteModel.js

```diff
@@ -0,0 +1,902 @@
+import _chunk from 'lodash/chunk';
+import _first from 'lodash/first';
+import _findIndex from 'lodash/findIndex';
+import _isNil from 'lodash/isNil';
+import _isString from 'lodash/isString';
+import _last from 'lodash/last';
+import _map from 'lodash/map';
+import _reduce from 'lodash/reduce';
+import _without from 'lodash/without';
+import LegModel from './LegModel';
+import BaseModel from '../../base/BaseModel';
+import NavigationLibrary from '../../navigationLibrary/NavigationLibrary';
+import AirportController from '../../airport/AirportController';
+import {
+    INVALID_INDEX,
+    INVALID_NUMBER,
+    REGEX
+} from '../../constants/globalConstants';
+import {
+    DIRECT_SEGMENT_DIVIDER,
+    PROCEDURE_OR_AIRWAY_SEGMENT_DIVIDER
+} from '../../constants/routeConstants';
+
+/**
+ * Representation of an aircraft's flight plan route
+ *
+ * This object contains all of the legs and waypoints the FMS will use to navigate.
+ * Each instance of an Aircraft has an FMS with a `RouteModel`, that it is able
+ * to modify, including adding/removing legs/waypoints, adding/removing waypoint
+ * restrictions, absorbing another `RouteModel`, etc.
+ *
+ * @class RouteModel
+ */
+export default class RouteModel extends BaseModel {
+    /**
+     * @for RouteModel
+     * @constructor
+     * @param navigationLibrary {NavigationLibrary}
+     * @param routeString {string}
+     */
+    constructor(navigationLibrary, routeString) {
+        if (!(navigationLibrary instanceof NavigationLibrary)) {
+            throw new TypeError(`Expected valid navigationLibrary, but received ${typeof navigationLibrary}`);
+        }
+
+        super();
```

Comment:
this should be the first line of the constructor.

Reply:
fixed, and verified this was the only such instance of that.

done